File: WebContent/WEB-INF/jython/contentspacearticle_query.py
from cn.edustar.jitar.data import BaseQuery
from base_action import *

class ContentSpaceArticleQuery (BaseQuery):
    def __init__(self, selectFields):
        BaseQuery.__init__(self, selectFields)
        self.params = ParamUtil(request)
        self.hasPicture = None
        self.contentSpaceId = None
        self.ownerType = None
        self.ownerId = None
        self.orderType = 0
        self.k = self.params.getStringParam("k")
        self.f = self.params.getStringParam("f")
        if self.f == None or self.f == "":
            self.f = "title"
        request.setAttribute("k", self.k)
        request.setAttribute("f", self.f)
    
    def initFromEntities(self, qctx):        
        qctx.addEntity("ContentSpaceArticle", "csa", "")
    
    def resolveEntity(self, qctx, entity):
        if "cs" == entity:
            qctx.addEntity("ContentSpace", "cs", "csa.contentSpaceId = cs.contentSpaceId")
        else:
            BaseQuery.resolveEntity(self, qctx, entity);

    def applyWhereCondition(self, qctx):
        if self.contentSpaceId != None:
            qctx.addAndWhere("(cs.contentSpaceId = :contentSpaceId OR cs.parentPath LIKE '%/"+str(self.contentSpaceId)+"/%')")
            qctx.setInteger("contentSpaceId", self.contentSpaceId)
        if self.ownerType != None:
            qctx.addAndWhere("cs.ownerType = :ownerType")
            qctx.setInteger("ownerType", self.ownerType)
        if self.ownerId != None:
            qctx.addAndWhere("cs.ownerId = :ownerId")
            qctx.setInteger("ownerId", self.ownerId)
        if self.hasPicture != None:
            if self.hasPicture == True:
                qctx.addAndWhere("cs.pictureUrl IS NOT NULL AND cs.pictureUrl <> ''")
            else:
                qctx.addAndWhere("cs.pictureUrl IS NULL OR cs.pictureUrl = ''")
            
            
        if self.k != None and self.f != None:
            newKey = self.k.replace("'","''").replace("%","[%]").replace("_","[_]").replace("[","[[]")
            if self.f == "createUserId":
                if self.k.isdigit():
                    qctx.addAndWhere("csa.createUserId = :createUserId")
                    qctx.setInteger("createUserId", int(newKey))
            else:
                qctx.addAndWhere("csa." + self.f + " LIKE :keyword")
                qctx.setString("keyword", "%" + newKey + "%")
                
    def applyOrderCondition(self, qctx):
        if self.orderType == 0:  
            qctx.addOrder("csa.contentSpaceArticleId DESC")

File: src/cn/edustar/jitar/service/ContentSpaceArticleQuery.java
package cn.edustar.jitar.service;

import cn.edustar.jitar.data.BaseQuery;
import cn.edustar.jitar.data.QueryContext;
import cn.edustar.jitar.util.CommonUtil;

public class ContentSpaceArticleQuery extends BaseQuery {

    public Integer contentSpaceId = null;
    public Integer ownerType = null;
    public Integer ownerId = null;
    public String k = null, f = null;
    public Integer orderType = 0;

    public ContentSpaceArticleQuery(String selectFields) {
        super(selectFields);
    }

    @Override
    public void initFromEntities(QueryContext qctx) {
        qctx.addEntity("ContentSpaceArticle", "csa", "");
    }

    public void resolveEntity(QueryContext qctx, String entity) {
        if ("cs".endsWith(entity)) {
            qctx.addEntity("ContentSpace", "cs", "csa.contentSpaceId = cs.contentSpaceId");
        } else {
            super.resolveEntity(qctx, entity);
        }
    }

    public void applyWhereCondition(QueryContext qctx) {
        if (this.contentSpaceId != null) {
            qctx.addAndWhere("(cs.contentSpaceId = :contentSpaceId OR cs.parentPath LIKE '%/" + this.contentSpaceId + "/%')");
            qctx.setInteger("contentSpaceId", this.contentSpaceId);
        }
        if (this.ownerType != null) {
            qctx.addAndWhere("cs.ownerType = :ownerType");
            qctx.setInteger("ownerType", this.ownerType);
        }
        if (this.ownerId != null) {
            qctx.addAndWhere("cs.ownerId = :ownerId");
            qctx.setInteger("ownerId", this.ownerId);
        }

        if (this.k != null && this.f != null) {
            String newKey = CommonUtil.escapeSQLString(this.k);
            if (this.f.equals("createUserId")) {
                if (CommonUtil.isInteger(this.k)) {
                    qctx.addAndWhere("csa.createUserId = :createUserId");
                    qctx.setInteger("createUserId", Integer.valueOf(this.k));
                }
            } else {
                qctx.addAndWhere("csa." + this.f + " LIKE :keyword");
                qctx.setString("keyword", "%" + newKey + "%");
            }
        }
    }
    public void applyOrderCondition(QueryContext qctx) {
        if (this.orderType == 0) {
            qctx.addOrder("csa.contentSpaceArticleId DESC");
        }
    }

}
